samcli/lib/utils/architecture.py

```diff
@@ -55,16 +35,26 @@ def validate_architecture_runtime(function: "Function") -> None:
     if function.packagetype == IMAGE:
         return
 
-    runtime_architectures = SUPPORTED_RUNTIMES.get(cast(str, function.runtime), [])
+    runtime_architectures = []
+    try:
+        runtime = Runtime.from_str(function.runtime)
+        runtime_architectures.extend(runtime.archs_as_list_of_str())
+    except ValueError:
+        LOG.debug("Unrecognized runtime %s", function.runtime)
```

Comment:
Should this be `debug` or `warning`/`error`?

Reply:
I updated it to `warning`. But I don't think it matters too much. At the end, if the runtime is unrecognized, this means `runtime_architectures` will be empty. And therefore L45 will always true and will raise `UnsupportedRuntimeArchitectureError`. 

And if no `Architecture` is provided, it will still fail with the unsupported runtime error like below:
```
Invoking bootstrap (providedi.al2023)                                                                                                                                                                                                                                                                                                                                                                                                                        
Error: Unsupported Lambda runtime: providedi.al2023. For a list of supported runtimes, please visit https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html
```